src/profile/components/multiItemEditor/MultiItemEditor.tsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+import to from 'await-to-js';
+import { Button, IconPlusCircle } from 'hds-react';
+
+import ProfileSection from '../../../common/profileSection/ProfileSection';
+import MultiItemRow from '../multiItemRow/MultiItemRow';
+import MultiItemAddressRow from '../multiItemAddressRow/MultiItemAddressRow';
+import commonFormStyles from '../../../common/cssHelpers/form.module.css';
+import useNotificationContent from '../editingNotifications/useNotificationContent';
+import EditingNotifications from '../editingNotifications/EditingNotifications';
+import {
+  useProfileDataEditor,
+  ActionListener,
+} from '../../hooks/useProfileDataEditor';
+import {
+  EditData,
+  EditDataType,
+  EditDataValue,
+  isNewItem,
+} from '../../helpers/editData';
+
+type Props = {
+  dataType: EditDataType;
+};
+
+export type RowItemProps = {
+  data: EditData;
+  onAction: ActionListener;
+  testId: string;
+  dataType: EditDataType;
+};
+
+function MultiItemEditor({ dataType }: Props): React.ReactElement | null {
+  const {
+    editDataList,
+    save,
+    reset,
+    add,
+    hasNew,
+    remove,
+  } = useProfileDataEditor({
+    dataType,
+  });
+  const { t } = useTranslation();
+  const {
+    content,
+    setErrorMessage,
+    setSuccessMessage,
+    clearMessage,
+  } = useNotificationContent();
+
+  const hasAddressList = dataType === 'addresses';
+  const isAddButtonDisabled = hasNew();
+  const RowComponent = hasAddressList ? MultiItemAddressRow : MultiItemRow;
+  const texts = (function() {
+    if (dataType === 'emails') {
```

Comment:
Hmm.. I wonder if there's a risk that this file becomes too complex, since we are trying to support all different dataTypes? If the requirements of different data types start changing a lot, it might be a good idea to apply some refactoring.

Reply:
Actually supporting only 3 data types. :) Those who have multiple items and have same editing tools and same notifications.

All this does it handle identical actions (save, add ,remove, set-primary), show notifications and render each item.

Reply:
Multiple items need a "parent / wrapper" that knows what children are doing. Later on, it must prevent children from being edited while re-ordering is in progress. Now it prevents multiple new items. 

"If the requirements of different data types start changing a lot, "

Then it should be addressed, perhaps not now?